### PR TITLE
Labs: GitHub-177: Add functionality to define constant to enable nginx logging

### PR DESCRIPTION
**Pull request by vishalkakadiya**
As discussed over the call on the **11th of January**:
- Adds functionality to define the **NGINX_HELPER_LOG** constant in the wp-config file to enable the logging feature.
- Users who are already using the nginx-helper plugin with **Enable Logging** will not be affected, but if they disable the logging then they need to define the **NGINX_HELPER_LOG** constant to enable the logging feature again.

**Ref:** https://github.com/rtCamp/labs/issues/177

### CODE REVIEW

**Review by gagan0123 — CHANGES_REQUESTED** [review]

File: admin/partials/nginx-helper-general-options.php [hunk 1]
```diff
@@ -539,10 +539,36 @@
...
 								<?php esc_html_e( 'Enable Logging', 'nginx-helper' ); ?>
 							</label>
+							<?php
+							if ( ! $is_checkbox_enabled ) {
+								printf(
+									'<pre>(%1$s <b>define( \'ENABLE_NGINX_HELPER_LOGS\', true )</b> %2$s <b>%3$s</b>)</pre>',
+									esc_html__( 'To enable the above checkbox, it needs to be defined the' ),
+									esc_html__( ' constant in the' ),
+									esc_html__( 'wp-config.php' )
+								);
```
[the remark this answers is not in the document]
Comment (vishalkakadiya) [reply admin/partials/nginx-helper-general-options.php]: Done

File: admin/partials/nginx-helper-general-options.php [hunk 2]
```diff
@@ -539,10 +539,36 @@
...
+
+							} elseif ( defined( 'ENABLE_NGINX_HELPER_LOGS' ) && true === ENABLE_NGINX_HELPER_LOGS ) {
+
+								$is_checkbox_enabled = true;
+							}
+							?>
+							<input
+								type="checkbox" value="1" id="enable_log" name="enable_log"
+								<?php checked( $nginx_helper_settings['enable_log'], 1 ); ?>
+								<?php echo esc_attr( $is_checkbox_enabled ? '' : ' disabled ' ); ?>
```
[the remark this answers is not in the document]
Comment (vishalkakadiya) [reply admin/partials/nginx-helper-general-options.php:551]: @gagan0123 This is fixed now. Thanks!

[the remark this answers is not in the document]
Comment (gagan0123) [reply admin/partials/nginx-helper-general-options.php:551]: Thanks :)

File: admin/partials/nginx-helper-general-options.php [hunk 3]
```diff
@@ -539,10 +539,29 @@
...
+								<?php checked( $nginx_helper_admin->is_nginx_log_enabled(), true ); ?>
+								<?php echo esc_attr( $is_checkbox_enabled ? '' : ' disabled ' ); ?>
+							/>
 							<label for="enable_log">
 								<?php esc_html_e( 'Enable Logging', 'nginx-helper' ); ?>
 							</label>
+							<?php
+							if ( ! $nginx_helper_admin->is_nginx_log_enabled() ) {
+								printf(
+									'<pre>[<b>NOTE:</b> %1$s<br /><b>define( \'NGINX_HELPER_LOG\', true )</b> %2$s <b>wp-config.php</b>]</pre>',
```
Comment (gagan0123) [inline admin/partials/nginx-helper-general-options.php:559]:
```suggestion
									'<pre>[<b>NOTE:</b> %1$s<br /><b> define( \'NGINX_HELPER_LOG\', true )</b> %2$s <b>wp-config.php</b>]</pre>',
```

File: tests/e2e-playwright/specs/04_enable-debug-option-test.spec.js [hunk 4]
```diff
@@ -16,18 +16,15 @@ test.describe("Enable the debug option", () => {
...
-      "Settings saved."
+    // Assert that the checkbox is disabled
+    await expect(page.locator("input#enable_log")).toBeDisabled();
+
+    expect(
+      page.locator(
+        "#post_form > div:nth-child(5) > div > table > tbody > tr:nth-child(1) > td > pre"
+      )
+    ).toHaveText(
+      "[NOTE: To activate the logging feature, you must define the constantdefine( 'NGINX_HELPER_LOG', true ) in your wp-config.php]"
```
Comment (gagan0123) [inline tests/e2e-playwright/specs/04_enable-debug-option-test.spec.js:27]:
```suggestion
      "[NOTE: To activate the logging feature, you must define the constant define( 'NGINX_HELPER_LOG', true ) in your wp-config.php]"
```

**Review by gagan0123 — CHANGES_REQUESTED** [review]

File: admin/partials/nginx-helper-general-options.php (hunk 3 above)
Comment (vishalkakadiya) [reply admin/partials/nginx-helper-general-options.php]: @gagan0123 fixed this now, thanks!

File: tests/e2e-playwright/specs/04_enable-debug-option-test.spec.js (hunk 4 above)
Comment (vishalkakadiya) [reply tests/e2e-playwright/specs/04_enable-debug-option-test.spec.js]: @gagan0123 fixed this now, thanks!

**Review by gagan0123 — APPROVED** [review]

**Pull request merged**